### ISSUE TITLE
4.0.0 pg_user_config.ip_filter_object does not set ips on aiven_pg

### ISSUE THREAD

**Issue by llicour** [aiven/terraform-provider-aiven#1073]
## What happened?
Following a deprecated warning on `ip_filter` after migrating from aiven provider 3.5.0 to 4.0.0, 
i change my code from
```
locals {
  aiven_static_ips = toset([
    "a.a.a.a/24",
    "b.b.b.b/24", 
  ])

resource "aiven_pg" "default" {
  ....
  pg_user_config {
    ip_filter = local.aiven_static_ips
  }
}
```
to 
```
resource "aiven_pg" "default" {
  ....
  pg_user_config {
    ip_filter_object {
      network     = "a.a.a.a/24"
      description = "some comment"
    }
    ip_filter_object {
      network     = "b.b.b.b/24"
      description = "some comment"
    }
  }
}
```

As effect, it remove all filters in aiven pg web interface, although there is no error or indication in `terraform apply`
Seems this append with a lot of filter applied

**Comment by Serpentiel** [comment]
hey, @llicour! 👋 

thanks for reporting this! I'll be taking a look now :)

**Comment by Serpentiel** [comment]
hey again!

I was not able to reproduce the issue, however, please make sure that you have your IP ranges in correct CIDR notation as defined in [RFC 4632](https://www.rfc-editor.org/rfc/rfc4632.html), meaning that if you want to allow a subnet of `/24` IP addresses, you should have the range ending with a leading zero, e.g. `1.1.1.0/24`

feel free to update the issue to include more detailed steps to reproduce the issue

**Comment by Conflit-de-Bus** [comment]
Hello,

I experienced a similar issue, with very similar code. But in my case, I had a lot more IPs that the example (around 15).

I did some additional tests. The details about my case:

- My pg service was created first, with `pg_user_config.ip_filter`
- I then changed the declaration to use multiple `pg_user_config.ip_filter_object` blocks

Apply: Same issue, the IPs "disappeared" in the console.

- I removed the `pg_user_config.ip_filter_object` blocks from my declaration.

Apply: no changes

- I added again all the `pg_user_config.ip_filter_object` blocks in my declaration

Apply: no changes

- I removed all the `pg_user_config.ip_filter_object` blocks in my declaration, but added only a few (3 or 4)

Apply: the declared IPs appeared in the console

I added little by little all the `pg_user_config.ip_filter_object` blocks that I needed, wich worked in the end, but is not ideal.

**Comment by Serpentiel** [comment]
hey, @Conflit-de-Bus! 👋

thank you for providing additional info, looks like this only occurs when migrating from `ip_filter` to `ip_filter_object` in a single step

I will take a look what could be done about it!

**Comment by Serpentiel** [comment]
hey, @llicour and @Conflit-de-Bus! 👋 

this has been fixed in v4.2.1

**Comment by Conflit-de-Bus** [comment]
Hi @Serpentiel 

Thank you, will test soon!
